### DISCUSSION TITLE
Notebooks - Is there an API that returns the markdown cell's state (editing or preview mode)?

### DISCUSSION THREAD

**Discussion by liorshk** [microsoft/vscode-discussions#1839] [Extension Development QnA] [answered]
I need a way to check if a notebook cell is in edit mode or preview mode. I found that this state is managed by the `getEditState()` function in the [notebookBrowser](https://github.com/microsoft/vscode/blob/0fa954530ca21e0bc455edbd8ac69ece9b1c0bdf/src/vs/workbench/contrib/notebook/browser/notebookBrowser.ts#L279):

```typescript
getEditState(): CellEditState;

export enum CellEditState {
    /**
     * Default state.
     * For markup cells, this is the rendered version of the markup.
     * For code cells, the browser focus is on the container instead of the editor.
     */
    Preview,

    /**
     * Editing mode. The source for markup or code is displayed in editors, and this state is persistent.
     */
    Editing
}
```

However, I’m not sure how to access `getEditState()` using the API.

One workaround I considered is using `onDidChangeNotebookDocument` and `onDidChangeNotebookEditorSelection` events, but this feels like a hack. 

Ultimately, I need a reliable way to detect when a user "saves" a markdown cell (essentially, when it switches from edit mode to preview mode). Any suggestions?

**Comment by isidorn** [comment]
@DonJayamanne might be able to help here

**Reply by DonJayamanne** [reply, marked as answer]
Unfortunately we do not have such a feature today.
